processing/src/main/java/org/apache/carbondata/processing/newflow/iterators/RecordReaderIterator.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.carbondata.processing.newflow.iterators;
+
+import java.io.IOException;
+
+import org.apache.carbondata.common.CarbonIterator;
+import org.apache.carbondata.common.logging.LogService;
+import org.apache.carbondata.common.logging.LogServiceFactory;
+
+import org.apache.hadoop.mapred.RecordReader;
+
+/**
+ * This iterator iterates RecordReader.
+ */
+public class RecordReaderIterator extends CarbonIterator<Object[]> {
```

Comment:
It is used for iterating RecordReader. I can move it to carbon-hadoop module but processing module need to be dependent on it. Already processing module is dependent on hadoop module so it becomes dependent on each other.